noobaa_sa/s3_client.py

```diff
@@ -0,0 +1,293 @@
+            else:
+                log.error(f"Failed to delete bucket {bucket_name}: {e}")
+                raise e
+        else:
+            log.info(f"Bucket {bucket_name} deleted successfully")
+
+    def list_buckets(self):
+        """
+        List buckets in an S3 account using boto3
+
+        Returns:
+            list: A list of the names of the buckets
+
+        """
+        log.info("Listing buckets via boto3")
+        response = self._boto3_client.list_buckets()
+        listed_buckets = [bucket_data["Name"] for bucket_data in response["Buckets"]]
+        log.info(f"Listed buckets: {listed_buckets}")
+        return listed_buckets
+
+    def list_objects(self, bucket_name, prefix="", use_v2=False):
+        """
+        List objects in an S3 bucket using boto3
+
+        Args:
+            bucket_name (str): The name of the bucket
+            prefix (str): A prefix where the objects will be listed from
+            use_v2 (bool): Whether to use list_objects_v2 instead of list_objects
+
+        Returns:
+            list: A list of the names of the objects
+
+        """
+        log.info(f"Listing objects in bucket {bucket_name} via boto3")
+        list_objects_method = (
+            self._boto3_client.list_objects_v2
+            if use_v2
+            else self._boto3_client.list_objects
+        )
+        output = list_objects_method(Bucket=bucket_name, Prefix=prefix)
+        listed_objects = []
+        if "Contents" in output:
+            listed_objects = [obj["Key"] for obj in output["Contents"]]
+        log.info(f"Listed objects: {listed_objects}")
+        return listed_objects
+
+    def put_object(self, bucket_name, object_key, object_data):
+        """
+        Put an object in an S3 bucket using boto3
+
+        """
+        log.info(f"Putting object {object_key} in bucket {bucket_name} via boto3")
+        self._boto3_client.put_object(
+            Bucket=bucket_name, Key=object_key, Body=object_data
+        )
+
+    def get_text_object_str(self, bucket_name, object_key):
+        """
+        Get the contents of a text object in an S3 bucket using boto3
+
+        For other types of objects, use get_object_contents
+
+        Returns:
+            str: The contents of the object
+
+        """
+        return self.get_object(bucket_name, object_key).read().decode("utf-8")
+
+    def get_object(self, bucket_name, object_key):
+        """
+        Get the contents of an object in an S3 bucket using boto3
+
+        Args:
+            bucket_name (str): The name of the bucket
+            object_key (str): The key of the object
+
+        Returns:
+            A botocore.response.StreamingBody object that can be read from
+
+        """
+        log.info(f"Getting object {object_key} from bucket {bucket_name} via boto3")
+        output = self._boto3_client.get_object(Bucket=bucket_name, Key=object_key)
+        return output["Body"]
```

Comment:
Please return entire `output` instead of `output["Body"]` so that user can play with output based on need

Reply:
done in https://github.com/red-hat-storage/noobaa-sa-ci/pull/11/commits/8af20a77685fcd906756287e38b5a7193d7c3dd2